### PR TITLE
Update spreading.dm

### LINKED ISSUE #15827: Pacid/Phenol Vines No Longer Eat Flooring / Wall Exteriors
#### Description of issue
Kudzu (or any vined plant for that matter) is no longer able to eat floors/walls.  This is likely due to the reagent change in Death Nettles from Pacid to Phenol.  I think Phenol is a child of pacid, so Idk why it doesnt work. 

#### Difference between expected and actual behavior
Phenol Vines destroying floor tiles, and exteriors of walls. 

#### Steps to reproduce 
Splice 100+ potency grass onto Death Nettles, Splice vines onto death nettles.
Plant Nettles
Wait for vines. 

#### Specific information for locating

	if(seed.chems && !isnull(seed.chems[PACID]))
			seed.chems[PACID] = null // Eating through the hull will make these plants completely inviable, albeit very dangerous.
			seed.chems -= null // Setting to null does not actually remove the entry, which is weird.

https://github.com/vgstation-coders/vgstation13/blob/246b9261f5df95639af1ec5c029a8e2fd9ed8136/code/modules/hydroponics/seed_controller.dm

#### Length of time in which bug has been known to occur
Probably since Phenol was added as the reagent for Death Nettles. 

#### Issue bingo
itaintworknomo

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by ghost**
Fixes #15827

:cl:
 * bugfix: Phenol kudzu now works again
 * tweak: Buffed kudzu growth speed
 * rscdel: Removed maximum growth radius for kudzu vines (NOT for creeper vines, which still have a maximum radius of 4)

### CODE REVIEW

**Comment by ghost** [comment]
Shit how do I add more changes

**Comment by Shadowmech88** [comment]
You can add more changes to your PR by pushing additional changes to the branch on your repo from which the PR was made.
In this case, `patch-1`.

**Comment by ghost** [comment]
How do I that with the online editor it just sends me to patch-2

**Comment by ghost** [comment]
getting there

**Comment by ghost** [comment]
There we go

**Comment by Exxion** [comment]
>creeping vines now grow as big as they want but grow 50% slower to compensate

That defeats like 70% of the purpose of creeping vines

**Comment by ghost** [comment]
They don't grab people, they don't block light, and they're tame as fuck, that's their purpose

No need to gimp them into 4 tiles max

**Comment by Exxion** [comment]
They also don't cover the whole station in obnoxious garbage
This changes them from useful fruit increasing tool to griff lite

**Comment by MadmanMartian** [comment]
If creeping vines are going to become limitless, might be time to make machetes

**Comment by ghost** [comment]
Scythes?

**Comment by MadmanMartian** [comment]
Just occurred to me that you can just make a sword, or a kitchen knife.

Also, scythes are order-only, aren't they? Except their mecha-mounted variant

**Comment by ghost** [comment]
Yeah, they come with the weedkiller grenades, too, and Cargo can also order goats

You can also make Plantbgone by hacking medbay's dispensers, or just light it on fire

**Comment by ghost** [comment]
Anyone having a problem with vines just aren't thinking of solutions. Yes, losing flamethrowers for an easy instant room clear was a sad loss but there's toxin in the medbay vendors to mix with 4 parts water, there's spray bottles, pepperspray, and extinguishers everywhere. Cargo and the silicons can get you goats, scythe's (which have a 3x3 aoe cut centered on itself) and weedkiller grenades. You can also just wrench some chairs and wall it off/weld the firelocks to contain it in the meantime. It wouldn't be half the problem people think it is if it wasn't so uncommon that people just defaulted to puttering at it with welders instead of one guy being able to spritz down an entire quadrant of the station by themselves in a jiffy.

Losing vineuloose in the transition to botany3 was a sad day.

**Comment by SonixApache** [comment]
sounds fantastic

There's also pneumatic scythes for the APLUs which I assume are immune to vines.

**Comment by Exxion** [comment]
I'm not even talking about vines. I'm talking about *creeping* vines. The tame vines that only spread four tiles and don't ensnare people, which are actually useful to increase fruit output without ruining the station.
The fact that you'd need goats or weedkiller to deal with them after the change is the problem. They're meant to be tame.

To be clear, I have no problem with making full vines spread forever. Just the creeping ones.

**Comment by MadmanMartian** [comment]
See if you can't find a value to scale the growth size off of. Potency'd be somewhat backwards as you can get high potency easy, and it's needed for getting high-yield crops.

Maybe growth speed?

**Comment by ghost** [comment]
Thinking about it better I can see Exxion's point, being able to cover botany in creepers with no chance of it spreading out is a nice feature. Then again yeah you can control that shit easily if it spreads.

In my opinion the best thing would be to have 3 stages of kudzu, creeper -> nolimit creeper -> superkudzu, but fuck me I don't know how to do that it took me like an hour to change 5 lines

**Comment by SonixApache** [comment]
so figure it out, spend some time on it

**Comment by N3X15** [comment]
Is this ready for merge?

**Comment by Exxion** [comment]
I'd say re-limit creeper and ship the rest of it and we can decide on creeper later, because there's also a bugfix in here

**Comment by ghost** [comment]
I gotta investigate this some more, I think there were actually 2 different "growth cap" mechanisms. One is the maximum radius and then there's ALSO a "max number of tiles".

If it does work that way I think I'll readd the creeper limit as number of tiles rather than radius, because that's allaround better

**Comment by ghost** [comment]
I was wrong, re-added maximum growth radius for creepers

**Comment by N3X15** [comment]
Let me know if this is ready.

**Comment by Unknown-as-Captain** [comment]
Full speed ahead captain

**Comment by ghost** [comment]
Full speed ahead captain, tested and works

**Comment by ghost** [comment]
Actually it seems walls were changed at some point, phenolkudzu can no longer get through them as they drop a girder 100% of the time and kudzu can't into girder

Before: 
glowing polyacid kudzu
After:
dreamseeker_2017-09-12_16-51-16

If you want you can merge this and I'll fix it in a different PR, will investigate

**Comment by ghost** [comment]
Duplicate or remodel the wall/floor check the phenol uses and include girders as a target too? That should be pretty quick to do.

If it's structured well enough, maybe start including a potency check for machinery, computers, airlocks, lights, apc/air alarm/pipes/wires/unary, loose items etc so that a dangerous and unchecked infestation in say, chemistry, could destroy the department instead of being a minor nuisance at best. In another pr of course.

File: code/modules/hydroponics/spreading/spreading_growth.dm [hunk 1]
```diff
@@ -124,6 +124,8 @@
 		for(var/obj/effect/plantsegment/neighbor in check_turf.contents)
 			neighbor.neighbors |= check_turf
 			plant_controller.add_plant(neighbor)
-	spawn(1) if(src) qdel(src) //fuck linebreaks amirite
```
Comment (Iamgoofball) [inline code/modules/hydroponics/spreading/spreading_growth.dm:127]: the fact that this was done intentionally is horrible

Comment (Iamgoofball) [reply code/modules/hydroponics/spreading/spreading_growth.dm:127]: only thing this is missing is to be using del and have a semicolon at the end

File: code/modules/hydroponics/spreading/vines.dm [hunk 2]
```diff
@@ -423,3 +423,4 @@
 			break
 
 	growth_queue = growth_queue + queue_end
+*/
```
Comment (Iamgoofball) [inline code/modules/hydroponics/spreading/vines.dm:426]: why not just delete this file if you're commenting it out entirely

**Comment by ghost** [comment]
That should do it

File: code/modules/hydroponics/spreading/spreading_growth.dm [hunk 3]
```diff
@@ -11,18 +11,19 @@
...
 			continue
-		if(locate(/obj/effect/plantsegment) in floor.contents)
+		if(locate(/obj/effect/plantsegment) in T.contents)
 			continue
-		if(floor.density)
+		if(T.density)
 			if(!isnull(seed.chems[PHENOL]))
-				spawn(rand(5,25)) floor.ex_act(3)
+				spawn(rand(5,25))
+					T.ex_act(prob(80) ? 3 : 2)
```
Comment (Iamgoofball) [inline code/modules/hydroponics/spreading/spreading_growth.dm]: stealth nerf?

Comment (ghost) [reply code/modules/hydroponics/spreading/spreading_growth.dm]: Doesn't leave a girder 20% of the time, so stealth buff, but I guess it's redundant now that they pass through girders anyways

File: code/modules/hydroponics/spreading/vines.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/modules/hydroponics/spreading/vines.dm:426]: https://github.com/vgstation-coders/vgstation13/pull/15971

**Pull request merged**